### PR TITLE
Lesson 02 Excercises

### PR DESCRIPTION
**Pull request by StrafingRun**
I've been having a bit of trouble figuring out how to submit. Let me know if this all looks right.

### CODE REVIEW

File: students/grant_dowell/Lesson_02/fibonacci.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@author: Grant Dowell
+
+Excercise 2.4 - Fibonacci Series
+"""
+
+def fibonacci(n):
+    """Return the nth value of the fibonacci sequence"""
+    if n == 0:
+        nth = 0
+    elif n ==1:
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fibonacci.py]: Missing whitespace after ==

File: students/grant_dowell/Lesson_02/fibonacci.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    """Return the nth value of the lucas sequence"""
+    if n == 0:
+        nth = 2
+    elif n == 1:
+        nth = 1
+    else:
+        nth = lucas(n-2) + lucas(n-1)
+    return nth
+
+def sum_series(n,n0=0,n1=1):
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fibonacci.py]: You'll want spaces after the commas here

File: students/grant_dowell/Lesson_02/fibonacci.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    n: The index of the sequence you are requesting.
+    n0=0: Optional parameter for the first value of the sequence.
+    n1=1: Optional parameter for the second value of the sequence.
+    """
+    if n == 0:
+        nth = n0
+    elif n == 1:
+        nth = n1
+    else:
+        nth = sum_series(n-2,n0,n1)+sum_series(n-1,n0,n1)
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fibonacci.py]: White spaces after commas here too

File: students/grant_dowell/Lesson_02/fibonacci.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#        print(fibonacci(n))
+        
+#    for n in range(8):
+ #       print(lucas(n))
+ 
+#     for n in range(8):
+#         print(sum_series(n))
+ 
+     for n in range(8):
+         print(sum_series(n,n0=2,n1=1))
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fibonacci.py]: And here.  Works well!

File: students/grant_dowell/Lesson_02/fizz_buzz.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+Created on Tue Dec 24 10:50:46 2019
+
+@author: Grant Dowell
+Excercise 2.3 - Fizz Buzz
+"""
+
+for n in range(100):
+    n = n+1
+    print_ln = ''
+    if n%3 == 0:
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fizz_buzz.py]: Missing whitespace around modulo operator, here as well below.

File: students/grant_dowell/Lesson_02/fizz_buzz.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        print_ln = print_ln+'Fizz'
+    if n%5 == 0:
+        print_ln = print_ln+'Buzz'
+    if print_ln == '':
+        print(n)
+    else:
+        print(print_ln)
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/fizz_buzz.py]: Works well!

File: students/grant_dowell/Lesson_02/grid_printer.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+grid_printer.py
+
+@author: Grant Dowell
+
+Excercise 2.2 - Grid Printer
+"""
+
+def print_grid(n,m):
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/grid_printer.py]: Whitespace after comma

File: students/grant_dowell/Lesson_02/grid_printer.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    def horizontal():
+        # Build String
+        #print_ln = plus+cell_size*minus+plus+cell_size*minus+plus
+        print_ln = plus
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/grid_printer.py]: Variable names should be clear: print_line for example

File: students/grant_dowell/Lesson_02/grid_printer.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        for c in range(n):
+            print_ln = print_ln+cell_size*minus+plus
+        print(print_ln)
+        
+    def vertical():
+        for p in range(cell_size):
+            #print_ln = bar+cell_size*space+bar+cell_size*space+bar
+            print_ln = bar
+            for c in range(n):
+                print_ln=print_ln+cell_size*space+bar
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/grid_printer.py]: Missing whitespace around assignment operator

File: students/grant_dowell/Lesson_02/grid_printer.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    for q in range(n):
+        vertical()
+        horizontal()
+
+            
+        
+if __name__ == '__main__':
+    print('******* Start Main *******')
+    
+    print_grid(3,4)
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/grid_printer.py]: Spaces after commas here and below

File: students/grant_dowell/Lesson_02/grid_printer.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    #print_grid(5,3)
+    print_grid(3.3,4.5)
+    print_grid(3.8,4.9)
+
```
Comment (charlesrobison) [inline students/grant_dowell/Lesson_02/grid_printer.py]: Looks good!

**Comment by StrafingRun** [comment]
Thanks for the feedback

**Pull request closed without merging**